File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap<Character , Integer> h = new HashMap<>();
        for(int i=0; i<s.length(); i++){
            char ch = s.charAt(i);
            h.put(ch , h.getOrDefault(ch , 0)+1);
        }
        for(int i=0; i<t.length(); i++){
            char ch = t.charAt(i);
            if(h.get(ch) != null){
                if(h.get(ch) == 1){
                    h.remove(ch);
                }else{
                    h.put(ch , h.get(ch) - 1);
                }
            }else{
                return false;
            }
        }
        return h.isEmpty();
    }
}

File: 0242-valid-anagram/0242-valid-anagram.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        if(s.Length != t.Length){
            return false;
        }
        int[] freq = new int[26];

        for(int i=0; i<s.Length; i++){
            char ch = s[i];
            freq[ch - 'a']++;
        }
        for(int i=0; i<t.Length; i++){
            char ch = t[i];
            freq[ch - 'a']--;
        }

        for(int i=0; i<26; i++){
           if(freq[i] != 0){
            return false;
           }
        }
        return true;
    }
}